adi/sync_start.py

```diff
@@ -99,3 +143,48 @@ def rx_sync_start_available(self):
             )
         except:  # noqa: E722
             return "arm"
+
+    @property
+    def rx_b_sync_start(self):
+        """ rx_b_sync_start: Issue a synchronisation request
+
+        Possible values are:
+            - **arm**: Writing this key will arm the trigger mechanism sensitive to an
+              external sync signal. Once the external sync signal goes high
+              it synchronizes channels within a ADC, and across multiple
+              instances. This bit has an effect only the EXT_SYNC
+              synthesis parameter is set.
+            - **disarm**: Writing this key will disarm the trigger mechanism
+              sensitive to an external sync signal. This bit has an
+              effect only the EXT_SYNC synthesis parameter is set.
+            - **trigger_manual**: Writing this key will issue an external sync event
+              if it is hooked up inside the fabric. This key has an effect
+              only the EXT_SYNC synthesis parameter is set.
+              This key self clears.
+        """
+        try:
+            return self._get_iio_dev_attr_str("sync_start_enable", _ctrl=self._rxadc2)
+        except:  # noqa: E722
+            if self._rxadc2.reg_read(0x68) & 1 == 1:
+                return "arm"
+            else:
+                return "disarm"
+
+    @rx_b_sync_start.setter
+    def rx_b_sync_start(self, value):
+        try:
+            self._set_iio_dev_attr_str("sync_start_enable", value, _ctrl=self._rxadc2)
+        except:  # noqa: E722
+            self._rxadc2.reg_write(0x48, 0x2)
+            while self._rxadc2.reg_read(0x68) == 0:
+                self._rxadc2.reg_write(0x48, 0x2)
+
+    @property
+    def rx_b_sync_start_available(self):
+        """ rx_sync_start_available: Returns a list of possible keys used for rx_sync_start """
+        try:
+            return self._get_iio_dev_attr_str(
+                "sync_start_enable_available", _ctrl=self._rxadc2
+            )
+        except:  # noqa: E722
```

Comment:
This tends to be really error prone. Can we put the actual exception that is expected?

Reply:
done